### PR TITLE
feat(passes): add pass dependency DAG with --deps and --dot flags

### PR DESCRIPTION
**Pull request by BANANASJIM**
## Summary
- Add `--deps` flag to `rdc passes` command to output render pass dependency DAG
- Infer dependencies from resource read/write patterns via `GetUsage()` per resource
- Support `--dot` flag for Graphviz DOT format visualization
- Support existing `--json` flag for programmatic consumers

## Algorithm
Uses the "usage_all scan" approach: iterate all resources, classify `ResourceUsage` events as reads/writes, bucket into passes by EID range, emit directed edges where `A.writes ∩ B.reads ≠ ∅`.

## Changes
- `src/rdc/services/query_service.py` — `build_pass_deps()` service function with read/write classification
- `src/rdc/handlers/query.py` — `_handle_pass_deps` daemon handler
- `src/rdc/commands/resources.py` — `--deps` and `--dot` flags on `passes` command
- `tests/unit/test_pass_deps.py` — 39 tests (service/daemon/CLI layers)

## Output Examples
```bash
# TSV (default)
rdc passes --deps
SRC     DST     RESOURCES
GBuffer Shadow  97,200

# Graphviz DOT
rdc passes --deps --dot | dot -Tpng -o deps.png

# JSON
rdc passes --deps --json
```

## Test plan
- 2390 tests passed, 94.06% coverage
- Service: 23 cases (graph structures, classification, edge cases)
- Daemon: 4 cases (happy path, errors, schema)
- CLI: 12 cases (TSV/DOT/JSON formats, error paths)

## Summary by CodeRabbit

* **New Features**
  * Added `--deps` flag to the passes command to show pass dependency DAG.
  * Added `--dot` flag for Graphviz DOT output (requires `--deps`).
  * Added `--graph` flag for a human-readable graph view (requires `--deps`).

* **Documentation**
  * Updated CLI quick reference with the new flags.

* **Tests**
  * Added comprehensive unit tests for pass dependency outputs (TSV/JSON/DOT/graph).

### CODE REVIEW

File: src/rdc/commands/resources.py [hunk 1]
```diff
@@ -189,9 +189,24 @@ def resource_cmd(resid: int, use_json: bool) -> None:
...
+    no_header: bool,
+    use_jsonl: bool,
+    quiet: bool,
+) -> None:
     """List render passes."""
+    if dot and not deps:
+        raise click.UsageError("--dot requires --deps")
+    if deps:
+        _passes_deps(use_json, dot)
+        return
```
[the remark this answers is not in the document]
Comment (BANANASJIM) [reply src/rdc/commands/resources.py:214]: Fixed — now raises `UsageError` when `--deps` is combined with `--no-header`, `--jsonl`, or `--quiet`.

File: src/rdc/services/query_service.py [hunk 2]
```diff
@@ -623,3 +627,122 @@ def get_pass_detail(
...
+    edges: list[dict[str, Any]] = []
+    for a in range(n):
+        if not writes[a]:
+            continue
+        for b in range(n):
+            if a == b:
+                continue
+            shared = writes[a] & reads[b]
+            if shared and (a, b) not in seen:
+                seen.add((a, b))
```
[the remark this answers is not in the document]
Comment (BANANASJIM) [reply src/rdc/services/query_service.py:733]: By design — this is intentionally not restricted to forward-only edges. OIT depth peeling captures have bidirectional pass dependencies (pass 2→3 and 3→2 ping-pong the same render targets). Enforcing `a < b` would hide real data-flow dependencies. The output represents actual resource read/write relationships, not a strict DAG.

File: src/rdc/services/query_service.py [hunk 3]
```diff
@@ -623,3 +627,122 @@ def get_pass_detail(
...
+                edges.append(
+                    {
+                        "src": passes[a]["name"],
+                        "dst": passes[b]["name"],
+                        "resources": sorted(shared),
+                    }
+                )
```
[the remark this answers is not in the document]
Comment (BANANASJIM) [reply src/rdc/services/query_service.py:740]: Same as above — bidirectional edges are intentional for ping-pong patterns (e.g. OIT depth peeling). The feature reports resource-level data flow, not a topologically sorted DAG.

File: src/rdc/services/query_service.py [hunk 4]
```diff
@@ -623,3 +627,122 @@ def get_pass_detail(
...
+            if a == b:
+                continue
+            shared = writes[a] & reads[b]
+            if shared and (a, b) not in seen:
+                seen.add((a, b))
+                edges.append(
+                    {
+                        "src": passes[a]["name"],
+                        "dst": passes[b]["name"],
+                        "resources": sorted(shared),
```
[the remark this answers is not in the document]
Comment (BANANASJIM) [reply src/rdc/services/query_service.py:738]: Acknowledged. In practice, `_friendly_pass_name()` appends sequential indices (e.g. "Colour Pass #1", "Colour Pass #2") so duplicates are rare. Adding `src_index`/`dst_index` would complicate the output for marginal benefit. Will revisit if real-world captures surface this issue.

File: src/rdc/commands/resources.py [hunk 5]
```diff
@@ -209,6 +228,83 @@ def passes_cmd(use_json: bool, no_header: bool, use_jsonl: bool, quiet: bool) ->
...
+
+
+def _format_dot(edges: list[dict[str, Any]]) -> None:
+    click.echo("digraph {")
+    for e in edges:
+        label = ",".join(str(r) for r in e["resources"])
+        src = e["src"].replace('"', '\\"')
+        dst = e["dst"].replace('"', '\\"')
+        click.echo(f'  "{src}" -> "{dst}" [label="{label}"];')
+    click.echo("}")
```
[the remark this answers is not in the document]
Comment (BANANASJIM) [reply src/rdc/commands/resources.py:258]: Fixed — backslashes are now escaped before double quotes in DOT output.
